pkg/autodiscovery/autoconfig.go

```diff
@@ -233,11 +212,15 @@ func (ac *AutoConfig) AddProvider(provider providers.ConfigProvider, shouldPoll
 	}
 
 	pd := &providerDescriptor{
-		provider: provider,
-		configs:  []integration.Config{},
-		poll:     shouldPoll,
+		provider:     provider,
+		configs:      []integration.Config{},
+		canPoll:      shouldPoll,
+		pollInterval: pollInterval,
 	}
 	ac.providers = append(ac.providers, pd)
+	if pd.canPoll {
```

Comment:
same here, I don't think this check is necessary, we should always start the provider